src/services/epg/epgClient.service.ts

```diff
@@ -18,25 +17,16 @@ export const isFulfilled = <T>(input: PromiseSettledResult<T>): input is Promise
   return false;
 };
 
-const epgProgramSchema = object().shape({
-  id: string().required(),
-  title: string().required(),
-  startTime: string()
-    .required()
-    .test((value) => (value ? isValid(new Date(value)) : false)),
-  endTime: string()
-    .required()
-    .test((value) => (value ? isValid(new Date(value)) : false)),
-  chapterPointCustomProperties: array().of(
-    object().shape({
-      key: string().required(),
-      value: string().test('required-but-empty', 'value is required', (value: unknown) => typeof value === 'string'),
-    }),
-  ),
-});
-
 @injectable()
-export default class EpgService {
+export default class EpgClientService {
+  private viewNexaProvider: EpgProviderService;
+  private jwProvider: EpgProviderService;
+
+  public constructor(@targetName(EPG_TYPE.JW) jwProvider: EpgProviderService, @targetName(EPG_TYPE.VIEW_NEXA) viewNexaProvider: EpgProviderService) {
+    this.viewNexaProvider = viewNexaProvider;
+    this.jwProvider = jwProvider;
+  }
```

Comment:
@dbudzins @ChristiaanScheermeijer not ideal, but maybe you have ideas how we could dynamically choose which provider to use depending on the media item.

Reply:
This can work the same as for the integration, right?

```ts
public loadProvider(item: PlaylistItem) {
  this.epgService = getModule(EpgService, getEpgName(item));
}
```

However, I would prefer something other than determining the EPG provider per media item but rather do it on the app config. With this change, only one EPG provider can be active at the time and can be loaded in the constructor instead.

The service modularization doesn't pay off completely if we still need to update the `epgClient` to add or replace certain services.

Ideally, when you want to implement a custom EPG service, only an update of the `container/register.ts` file would be needed 😄

Reply:
Yeah, that was my first intention to use `getModule` in the client class, but I would still prefer a native way of DI than using this trick. App Config definition could be an option, though in this case both implementations can't be used together (will clarify if it could be a problem).

Reply:
It is a possible use case when a customer adds two channels of different providers into one playlist.
So we will have to keep it on the media level.

Reply:
In this case, it would still make sense to use the same mechanism as that used for integrations. But ... currently, the same issue exists there;

If you want to introduce your own EPG provider or Integration, you must make changes in multiple files. I'm hoping we can find a solution that only requires adding the service, and the rest of the application knows how to process this.

I'm thinking about making the integration configuration extendable. Something like this:

```ts
type Config = {
  ...
  integrations: {
    cleeng?: Cleeng;
    jwp?: JWP;
    [key: string]: unknown;
  };
};
```

We know what to expect from the Cleeng and JWP configurations, but when integrating Adobe, we don't know what properties are needed. Validation of these properties should be done in the service.

The rest of the application shouldn't deal with the integration configuration. The integration services should provide all the information needed by the application.

Because the `integrations` type contains basically a wildcard for any configuration, we could look up the first key in the `integrations` object and use that as the "named resource".

This would also work for EPG where the `scheduleType` or `providerName` is used to request the correct EPG provider with a fallback to the JWP provider.

Reply:
Used multiInject decorator here.